InvenTree/order/views.py

```diff
@@ -995,6 +996,7 @@ def post(self, request, *args, **kwargs):
 
                 part.order_supplier = supplier_part.id if supplier_part else None
                 part.order_quantity = quantity
+                part.purchase_price = supplier_part.get_price(quantity) / quantity if supplier_part else None
```

Comment:
Same here, `quantity` can be 0 and `get_price` can return `None`

Reply:
quantity can not be 0 (line 992), price is handeld now